### PR TITLE
[Improve] remove checkstyle and scalastyle check

### PR DESCRIPTION
**Pull request by liugddx**
*(Please specify the module before the PR name: [core] ... or [flink] ...)*

### Purpose

We already have spotless to format the code, so remove the check style plugin.

### Tests

*(List UT and IT cases to verify this change)*

### API and Format 

*(Does this change affect API or storage format)*

### Documentation

*(Does this change introduce a new feature)*

### CODE REVIEW

**Review by zhuangchong — CHANGES_REQUESTED** [review]
checkstyle should not be removed. Currently, the project uses spotless format and checkstyle together.
For example: `import *`, `shade use`, etc. in checkstyle

**Comment by liugddx** [comment]
> checkstyle should not be removed. Currently, the project uses spotless format and checkstyle together. For example: `import *`, `shade use`, etc. in checkstyle

https://www.libhunt.com/compare-spotless-vs-checkstyle?ref=compare

I think spotless and checkstyle are duplicate checks, we should not use them at the same time, If spotless doesn't support these checks, what we should do is make it support these checks.

**Comment by zhuangchong** [comment]
> > checkstyle should not be removed. Currently, the project uses spotless format and checkstyle together. For example: `import *`, `shade use`, etc. in checkstyle
> 
> https://www.libhunt.com/compare-spotless-vs-checkstyle?ref=compare
> 
> I think spotless and checkstyle are duplicate checks, we should not use them at the same time, If spotless doesn't support these checks, what we should do is make it support these checks.

When removing checkstyle, can you add the project checkstyle special configuration item to the spotless format?

**Comment by liugddx** [comment]
> > > checkstyle should not be removed. Currently, the project uses spotless format and checkstyle together. For example: `import *`, `shade use`, etc. in checkstyle
> > 
> > 
> > https://www.libhunt.com/compare-spotless-vs-checkstyle?ref=compare
> > I think spotless and checkstyle are duplicate checks, we should not use them at the same time, If spotless doesn't support these checks, what we should do is make it support these checks.
> 
> When removing checkstyle, can you add the project checkstyle special configuration item to the spotless format?

I would love to try.  Thank you so much for your suggestion.

**Comment by liugddx** [comment]
> > > > checkstyle should not be removed. Currently, the project uses spotless format and checkstyle together. For example: `import *`, `shade use`, etc. in checkstyle
> > > 
> > > 
> > > https://www.libhunt.com/compare-spotless-vs-checkstyle?ref=compare
> > > I think spotless and checkstyle are duplicate checks, we should not use them at the same time, If spotless doesn't support these checks, what we should do is make it support these checks.
> > 
> > 
> > When removing checkstyle, can you add the project checkstyle special configuration item to the spotless format?
> 
> I would love to try. Thank you so much for your suggestion.

I added import wildcard rules, PTAL thanks  @zhuangchong

File: pom.xml [hunk 1]
```diff
@@ -798,7 +763,11 @@ under the License.
                             <importOrder>
                                 <order>org.apache.paimon,org.apache.paimon.shaded,,javax,java,scala,\#</order>
                             </importOrder>
-
+                            <replaceRegex>
+                                <name>Remove wildcard imports</name>
+                                <searchRegex>import\s+(static)*\s*[^\*\s]+\*;(\r\n|\r|\n)</searchRegex>
+                                <replacement>$1</replacement>
+                            </replaceRegex>
```
Comment (zhuangchong) [inline pom.xml:770]: This is not a very good solution, the part is to remove the import * if I understand correctly. If this is the case, after the contributor executes `mvn spotless:apply`, he submits the PR, and only when he executes the CI will he find that some classes are not imported.

**Review by zhuangchong — COMMENTED** [review]
Let's see what everyone thinks about this change.

File: pom.xml (hunk 1 above)
Comment (liugddx) [reply pom.xml:770]: I'm afraid there is no good solution yet.
https://github.com/diffplug/spotless/issues/649.  Perhaps it is more appropriate now, thanks for help and i will close this pr . @zhuangchong

**Pull request closed without merging**